### PR TITLE
Initial addition of category tags

### PR DESCRIPTION
**Pull request by clone1018**
- [x] Refactor `ChannelClassification`
- [x] Add tags to the API 
- [x] Add tests for the API tags
- [ ] Store tags inside the `Stream` whenever a channel goes live
- [x] Add additional tests for any missing functions
- [ ] Make it so `count_usage` is only incremented on start_stream
- [ ] Use `slug` instead of `name` in identifier

### CODE REVIEW

**Comment by clone1018** [comment]
I really want to split out the Category & Tag functionality in the Streams context into it's own `ChannelClassifier` or something.

File: lib/glimesh/streams/stream.ex [hunk 1]
```diff
@@ -23,6 +23,9 @@ defmodule Glimesh.Streams.Stream do
     field :new_subscribers, :integer
     field :resub_subscribers, :integer
 
+    field :global_tags, {:array, :integer}
+    field :category_tags, {:array, :integer}
```
Comment (clone1018) [inline lib/glimesh/streams/stream.ex:27]: These are not used yet.

File: lib/glimesh/streams/category.ex [hunk 2]
```diff
@@ -7,37 +7,20 @@ defmodule Glimesh.Streams.Category do
 
   schema "categories" do
     field :name, :string
-    field :tag_name, :string
     field :slug, :string
-    # field :parent_id, :integer
-    field :avatar, Glimesh.CategoryAvatar.Type
-    belongs_to :parent, Glimesh.Streams.Category
```
Comment (clone1018) [inline lib/glimesh/streams/category.ex:14]: Removing all the old avatar / parent category stuff.

File: lib/glimesh/streams/tag.ex [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  def set_identifier_attribute(changeset) do
+    category_name =
+      if category_id = get_field(changeset, :category_id) do
+        Glimesh.ChannelCategories.get_category_by_id!(category_id).name
+      else
+        "Global"
+      end
+
+    name = get_field(changeset, :name)
```
Comment (clone1018) [inline lib/glimesh/streams/tag.ex:42]: This needs to be adjusted in the future to use the Slug so it dedupes more similar entries

File: lib/glimesh_web/live/streams_live/list.ex [hunk 4]
```diff
@@ -24,26 +27,33 @@ defmodule GlimeshWeb.StreamsLive.List do
...
       nil ->
         {:ok, redirect(socket, to: "/")}
     end
   end
 
-  def mount(_, _, socket) do
-    {:ok, redirect(socket, to: "/")}
+  @impl true
+  def handle_params(params, _uri, socket) do
+    channels = ChannelLookups.filter_live_channels(Map.take(params, ["category", "tag"]))
```
Comment (clone1018) [inline lib/glimesh_web/live/streams_live/list.ex:52]: This is a super neat way of handling page loads, where `mount` isn't called twice. So when you are filtering streams it happens virtually immediately.

File: lib/glimesh_web/live/user_settings/components/channel_settings_live.html.leex [hunk 5]
```diff
@@ -1,8 +1,8 @@
-<%= form_for @channel_changeset, @route, [multipart: true, class: "form"], fn f -> %>
+<%= f = form_for @channel_changeset, @route, [phx_change: :change_channel, multipart: true, class: "form"] %>
```
Comment (clone1018) [inline lib/glimesh_web/live/user_settings/components/channel_settings_live.html.leex:1]: Can't use anonymous forms with child components

File: priv/repo/migrations/20210124122616_create_tags.exs [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+  import Ecto.Query, warn: false
+
+  def change do
+    create table(:tags) do
+      add :identifier, :string
+      add :name, :string
+      add :slug, :string
+
+      add :icon, :string
```
Comment (clone1018) [inline priv/repo/migrations/20210124122616_create_tags.exs:12]: To be used later

File: lib/glimesh_web/live/user_live/components/tag_selector.ex [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  def render(assigns) do
+    ~L"""
+    <div id="tagify" phx-update="replace">
+      <%= text_input @form, @field,
+        value: @existing_tags,
+        class: "tagify",
+        "data-category": @current_category_id,
+        "data-tags": @tags,
+        "phx-hook": "TagSelector",
+        placeholder: "describe your stream" %>
```
Comment (mja00) [inline lib/glimesh_web/live/user_live/components/tag_selector.ex:14]: Should this be gettext'd? I'm running at like 2% brain atm.

**Review by mja00 — APPROVED** [review]
Looks good to me.

File: lib/glimesh_web/live/user_live/components/tag_selector.ex (hunk 7 above)
Comment (clone1018) [reply lib/glimesh_web/live/user_live/components/tag_selector.ex:14]: Yeah, will add

**Pull request merged**